core/classes/Adapter.js

```diff
@@ -117,6 +132,11 @@ module.exports = class Adapter {
 
     const fixedData = this._sanitize(identity, dataToInsert)
 
+    const validation = this._validate(this.schemas[identity], _.omitBy(fixedData, _.isNil))
+    if (!validation.passed) {
+      throw new Error(`Property ${validation.errors[0].dataPath} validation fail: ${validation.errors[0].message}`)
```

Comment:
Please ensure this message works well for the case of missing required property.